dev/ci/preflight.sh

```
@@ -126,8 +126,18 @@ else
-    # ~instant.  --dry-run --Werror returns non-zero on any reformatting.
-    if uvx clang-format@"$CF_VER" --dry-run --Werror $ALL_CPP 2>&1 | grep -q .; then
+    # ~instant.  --dry-run --Werror prints one diagnostic per violation AND
+    # exits non-zero.
+    #
+    # Capture the output and test that, rather than piping into `grep -q`:
+    # under `set -o pipefail` (line 35) the pipeline inherits clang-format's
+    # non-zero exit, so `if <pipeline>` was false exactly when violations
+    # existed and the gate reported PASS.  With no violations clang-format
+    # exits 0 but grep finds nothing and exits 1 -- also false, also PASS.
+    # Both branches led to PASS, so the gate could never fail.
+    CF_OUT="$(uvx clang-format@"$CF_VER" --dry-run --Werror $ALL_CPP 2>&1 || true)"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)